components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/token/DefaultApiKeyGenerator.java

```diff
@@ -94,9 +96,16 @@ protected String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) {
         if (expireIn != -1) {
             jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.EXPIRY_TIME, expireIn);
         }
-        jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.SUBSCRIBED_APIS, jwtTokenInfoDTO.getSubscribedApiDTOList());
-        jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.TIER_INFO, jwtTokenInfoDTO.getSubscriptionPolicyDTOList());
-        jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.APPLICATION, jwtTokenInfoDTO.getApplication());
+        if (isLightweightAPIKeyGenerationEnabled()) {
+            jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.APPLICATION_ID,
+                    jwtTokenInfoDTO.getApplication().getId());
+        } else {
+            jwtClaimsSetBuilder.claim(APIConstants.JwtTokenConstants.SUBSCRIBED_APIS,
```

Comment:
Is it possible to skip the db calls executed to fetch these information when LW key is enabled? We don't need those in that flow, right?

Reply:
exclude DB calls